src/adapter/tests/timestamp_selection.rs

```diff
@@ -261,6 +264,8 @@ fn parse_query_when(s: &str) -> QueryWhen {
 /// Transaction isolation can also be set. The `determine` directive runs determine_timestamp and
 /// returns the chosen timestamp. Append `full` as an argument to it to see the entire
 /// TimestampDetermination.
+// allow `futures::block_on` for testing.
```

Comment:
this turns out to be a pretty big footgun in practice, even for tests. highly recommend figuring out a way to use tokio's block_on